shub/schedule.py

```diff
@@ -78,17 +77,17 @@ def cli(spider, argument, set, priority, units, tag):
 
 def schedule_spider(project, endpoint, apikey, spider, arguments=(), settings=(),
                     priority=DEFAULT_PRIORITY, units=None, tag=()):
-    conn = Connection(apikey, url=endpoint)
+    client = ScrapinghubClient(apikey, dash_endpoint=endpoint)
     try:
-        kw = dict(x.split('=', 1) for x in arguments)
-        if units is not None:
-            kw['units'] = units
-        return conn[project].schedule(
-            spider,
-            job_settings=json.dumps(dict(x.split('=', 1) for x in settings)),
+        project = client.get_project(project)
+        job = project.jobs.run(
```

Comment:
This is not something well-known but currently used API allows passing `cmd_args` and `meta` within `arguments`. New client supports it as well, via corresponding fields for `jobs.run` function. Command args field works as expected,
```
shub schedule py:testargs.py -a cmd_args='["val1","val2"]'
```
and we have plans to extend meta field usage, so let's keep it for now in the sake of backward compatibility. 

I mean something like:
```
arguments = dict(x.split('=', 1) for x in arguments)
cmd_args = arguments.pop('cmd_args', None)
meta = arguments.pop('meta', None)
job = project.jobs.run(
    ...
    job_args=arguments,
    meta=json.loads(meta) if meta or {},
    cmd_args=json.loads(cmd_args) if cmd_args or [],
    ...
)
```

Reply:
Hi Viktor! Sorry for the delay.

I updated this part of the code, but I'm sending `cmd_args` as a string instead of a list. The [documentation for `Jobs.run`](https://python-scrapinghub.readthedocs.io/en/latest/client/apidocs.html#scrapinghub.client.jobs.Jobs.run) says `cmd_args` is a "string with script command args", and I did a small test with the following script:

```
import logging
import sys

logging.basicConfig(level=logging.DEBUG)

if __name__ == '__main__':
    logging.info(sys.argv)

```
with both `cmd_args=json.loads(cmd_args) if cmd_args else [],` and `cmd_args=cmd_args`, I found that sending a string gets all of the values:
```
$ shub log -f 350109/3/6
2018-12-13 17:44:36.311000 INFO Log opened.
2018-12-13 17:44:36.627000 INFO [root] ['show_args.py', '[val1,val2]']
```

but sending a parsed list gets only the last value:
```
$ shub log -f 350109/3/7
2018-12-13 17:45:36.082000 INFO Log opened.
2018-12-13 17:45:36.404000 INFO [root] ['show_args.py', 'val2']
```

Reply:
Hi Viktor! Sorry for the delay.

I updated this part of the code, but I'm sending `cmd_args` as a string instead of a list. The [documentation for `Jobs.run`](https://python-scrapinghub.readthedocs.io/en/latest/client/apidocs.html#scrapinghub.client.jobs.Jobs.run) says `cmd_args` is a "string with script command args", and I did a small test with the following script:

```
import logging
import sys

logging.basicConfig(level=logging.DEBUG)

if __name__ == '__main__':
    logging.info(sys.argv)

```
with both `cmd_args=json.loads(cmd_args) if cmd_args else [],` and `cmd_args=cmd_args`, I found that sending a string gets all of the values:
```
$ shub log -f xxxxxx/3/6
2018-12-13 17:44:36.311000 INFO Log opened.
2018-12-13 17:44:36.627000 INFO [root] ['show_args.py', '[val1,val2]']
```

but sending a parsed list gets only the last value:
```
$ shub log -f xxxxxx/3/7
2018-12-13 17:45:36.082000 INFO Log opened.
2018-12-13 17:45:36.404000 INFO [root] ['show_args.py', 'val2']
```

Reply:
Oh, you're right indeed :+1: The endpoint expects a string there, and I also messed with the format, so it should be used as `shub schedule py:script.py -a cmd_args="--flag1 --flag2=0 val1 val2"` for proper args conversion.